### PR TITLE
Add unit tests for ShardingRenameTableStatementValidator (#14726)

### LINKED ISSUE #14726: Add more unit test for ShardingRenameTableStatementValidator
Hi, community:

This issue is to add more unit tests for the ShardingRenameTableStatementValidator class. Welcome to claim them.

The specific subtasks are as follows:

- [ ]  Add unit test for ShardingRenameTableStatementValidator#preValidate
- [ ]  Add unit test for ShardingRenameTableStatementValidator#postValidate

Through this issue, you can understand the process of participating in Apache ShardingSphere, and be familiar with sharding validate.

### PR DESCRIPTION
**Pull request by homeway88**
Fixes #14726.

Changes proposed in this pull request:
- Add unit test for ShardingRenameTableStatementValidator#preValidate
- Add unit test for ShardingRenameTableStatementValidator#postValidate

### CODE REVIEW

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/ShardingRenameTableStatementValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    @Test
+    public void assertPreValidateNormalCase() {
+        SQLStatementContext<RenameTableStatement> sqlStatementContext = createRenameTableStatementContext("t_not_sharding_table", "t_not_sharding_table_new");
+        ShardingSphereSchema schema = mock(ShardingSphereSchema.class);
+        new ShardingRenameTableStatementValidator().preValidate(shardingRule, sqlStatementContext, Collections.emptyList(), schema);
+    }
+
+    @Test(expected = ShardingSphereException.class)
+    public void assertPostValidateDifferentRouteUnitsAndDataNodesSize() {
+        // route units and actual data nodes are different size
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/ShardingRenameTableStatementValidatorTest.java:81]: @homeway88 Please remove this comment here.

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/ShardingRenameTableStatementValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        when(shardingRule.isShardingTable("t_order")).thenReturn(true);
+        SQLStatementContext<RenameTableStatement> sqlStatementContext = createRenameTableStatementContext("t_order", "t_user_order");
+        ShardingSphereSchema schema = mock(ShardingSphereSchema.class);
+        ConfigurationProperties props = mock(ConfigurationProperties.class);
+        new ShardingRenameTableStatementValidator().postValidate(shardingRule, sqlStatementContext, Collections.emptyList(), schema, props, routeContext);
+    }
+
+    @Test
+    public void assertPostValidateNormalCase() {
+        // route units and actual data nodes are equal size
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/ShardingRenameTableStatementValidatorTest.java:96]: @homeway88 Please remove this comment here.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/ShardingRenameTableStatementValidatorTest.java (hunk 1 above)
Comment (homeway88) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/ShardingRenameTableStatementValidatorTest.java]: Done

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/ShardingRenameTableStatementValidatorTest.java (hunk 2 above)
Comment (homeway88) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/test/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/ShardingRenameTableStatementValidatorTest.java]: Done

**Review by strongduanmu — APPROVED** [review]
@homeway88 Thank you.

**Pull request merged**